eng/common/scripts/artifact-metadata-parsing.ps1

```diff
@@ -0,0 +1,469 @@
+
+  try {
+    $nugetVersions = Invoke-RestMethod -Method "GET" -Uri $nugetUri
+
+    return $nugetVersions.versions.Contains($pkgVersion)
+  }
+  catch {
+    $statusCode = $_.Exception.Response.StatusCode.value__
+    $statusDescription = $_.Exception.Response.StatusDescription
+
+    # if this is 404ing, then this pkg has never been published before
+    if ($statusCode -eq 404) {
+      return $False
+    }
+
+    Write-Host "Nuget Invocation failed:"
+    Write-Host "StatusCode:" $statusCode
+    Write-Host "StatusDescription:" $statusDescription
+    exit(1)
+  }
+
+}
+
+# Parse out package publishing information given a python sdist of ZIP format.
+function ParsePyPIPackage($pkg, $workingDirectory) {
+  $pkg.Basename -match $SDIST_PACKAGE_REGEX | Out-Null
+
+  $pkgId = $matches["package"]
+  $pkgVersion = $matches["versionstring"]
+
+  $workFolder = "$workingDirectory$($pkg.Basename)"
+  $origFolder = Get-Location
+  mkdir $workFolder
+
+  Expand-Archive -Path $pkg -DestinationPath $workFolder
+  $releaseNotes = &"${PSScriptRoot}/../Extract-ReleaseNotes.ps1" -ChangeLogLocation @(Get-ChildItem -Path $workFolder -Recurse -Include "CHANGELOG.md")[0]
+  $readmeContentLoc = @(Get-ChildItem -Path $workFolder -Recurse -Include "README.md")[0]
+  if (Test-Path -Path $readmeContentLoc) {
+    $readmeContent = Get-Content -Raw $readmeContentLoc
+  }
+  Remove-Item $workFolder -Force  -Recurse -ErrorAction SilentlyContinue
+
+  return New-Object PSObject -Property @{
+    PackageId      = $pkgId
+    PackageVersion = $pkgVersion
+    Deployable     = $forceCreate -or !(IsPythonPackageVersionPublished -pkgId $pkgId -pkgVersion $pkgVersion)
+    ReleaseNotes   = $releaseNotes
+    ReadmeContent  = $readmeContent
+  }
+}
+
+function ParseCArtifact($pkg, $workingDirectory) {
+  $packageInfo = Get-Content -Raw -Path $pkg | ConvertFrom-JSON
+  $packageArtifactLocation = (Get-ItemProperty $pkg).Directory.FullName
+
+  $releaseNotes = ExtractReleaseNotes -changeLogLocation @(Get-ChildItem -Path $packageArtifactLocation -Recurse -Include "CHANGELOG.md")[0]
+
+  $readmeContentLoc = @(Get-ChildItem -Path $packageArtifactLocation -Recurse -Include "README.md")[0]
+  if (Test-Path -Path $readmeContentLoc) {
+    $readmeContent = Get-Content -Raw $readmeContentLoc
+  }
+
+  return New-Object PSObject -Property @{
+    PackageId      = $packageInfo.name
+    PackageVersion = $packageInfo.version
+    # Artifact info is always considered deployable for C becasue it is not
+    # deployed anywhere. Dealing with duplicate tags happens downstream in
+    # CheckArtifactShaAgainstTagsList
+    Deployable     = $true
+    ReleaseNotes   = $releaseNotes
+    ReadmeContent  = $readmeContent
+  }
+}
+
+# Returns the pypi publish status of a package id and version.
+function IsPythonPackageVersionPublished($pkgId, $pkgVersion) {
+  try {
+    $existingVersion = (Invoke-RestMethod -Method "Get" -Uri "https://pypi.org/pypi/$pkgId/$pkgVersion/json").info.version
```

Comment:
Should we use our wrapper Invoke-RestMethod.

Reply:
Actually. Yes. I'll update it.